diff --git a/frontend/src/components/ProcessForm.vue b/frontend/src/components/ProcessForm.vue
--- a/frontend/src/components/ProcessForm.vue
+++ b/frontend/src/components/ProcessForm.vue
@@ -106,2 +106,20 @@
+    <!-- Process Credentials Field (Python Only) -->
+    <div class="flex items-center" v-if="editedProcess.target_type == 'python'">
+      <label for="credentials_id" class="w-1/5 font-semibold">Process credentials</label>
+      <div class="w-full">
+        <select
+          id="credentials_id"
+          v-model="editedProcess.credentials_id"
+          class="select select-bordered w-full"
+        >
+          <option :value="null" :key="0">None</option>
+          <option :value="cred.id" v-for="cred in credentials" :key="cred.id">
+            {{ cred.name }}
+          </option>
+        </select>
+        <small class="text-base-content/60 block mt-1">Credential used by the automation</small>
+      </div>
+    </div>
+
     <!-- Unsupported Platform Message -->
     <div class="text-center text-base-content/60" v-if="editedProcess.target_type != 'python'">
